website/docs/language/state/sensitive-data.mdx

```diff
@@ -6,20 +6,24 @@ description: Sensitive data in Terraform state.
 # Sensitive Data in State
 
 Terraform state can contain sensitive data, depending on the resources in use
-and your definition of "sensitive." The state contains resource IDs and all
-resource attributes. For resources such as databases, this may contain initial
+and your definition of "sensitive." Unless your variables or resources are `ephemeral`, the state contains resource IDs and all resource attributes. For resources such as databases, this can contain initial
 passwords.
 
 When using local state, state is stored in plain-text JSON files.
 
-When using [remote state](/terraform/language/state/remote), state is only ever held in
-memory when used by Terraform. It may be encrypted at rest, but this depends on
-the specific remote state backend.
+When using [remote state](/terraform/language/state/remote), state is only ever held in memory when used by Terraform. It may be encrypted at rest, but this depends on the specific remote state backend.
+
+## Ephemeral data
+
+-> **Note**: Ephemeral variables, outputs, and resources are in public beta and available in Terraform v1.10 and later. The workflows of ephemeral values are subject to change.
+
+Terraform allows you to mark variables, locals, outputs, and resources as ephemeral.  Marking a block with the `ephemeral` attribute tells Terraform that this data is available during runtime, but Terraform does not write that data to state or plan files. The `ephemeral` property is helpful when managing credentials, tokens, or other temporary resources you do not want to store in Terraform state files.
```

Comment:
"Terraform allows you to mark variables, locals, outputs, _and resources as ephemeral_."

Not sure this was intended, but the latter part of this sentence seems to imply that you can mark existing `resource` blocks as ephemeral, rather than it being a completely separate concept that providers need to explicitly support.